### ISSUE TITLE
YouTube and Wikipedia links that fail

### ISSUE THREAD

**Issue by mcognetta** [ncase/nutshell#13]
I encountered several YouTube and Wikipedia links that fail to load on Nutshell (both on my site and in your example sandbox). I would guess this affects other types of links, but I include these two since they are explicitly given as examples on your site. As a note: all of the examples in your example sandbox worked fine on my site.

I simply don't know enough to know if this is an issue with Nutshell or a problem with the specific pages that I tried to load or something else. Hope this can shed some light on it.

P.S. (so that it doesn't get lost at the bottom) This project is awesome, it is *just* what I've been looking for for a while.

------------------------------------

### YouTube:
```markdown
[:julia](https://www.youtube.com/watch?v=N4h46_TCmGc&t=10s)
```
![nutshell_julia_youtube](https://user-images.githubusercontent.com/6856636/185728815-4f4bafdf-c75e-4443-8d92-883fa000edeb.png)

Clicking on the video produces an error and then redirects to YouTube's main page. Embedding without the timestamp works as expected.

--------------------------------------------------------------------

### Wikipedia:
```markdown
[:Miller](https://en.wikipedia.org/wiki/Miller–Rabin_primality_test)
```
![nutshell_wikipedia](https://user-images.githubusercontent.com/6856636/185728867-000dc0a0-bcd0-4f5e-bb4b-7beb9dac5ed6.png)

This also happens with 
```markdown
[:Miller](https://en.wikipedia.org/wiki/Miller%E2%80%93Rabin_primality_test)
```

------------------------------------------------
### Other notes:
I thought that it may be due to the character encoding during HTML generation (`–` changed to `%E2%80%93` and `&` changed to `&amp;`), but I tried writing it directly with HTML and it still didn't work.

**Comment by ncase** [comment]
Thank you for catching both of these bugs, with detailed testable examples @mcognetta ! I fixed both of them with the new Nutshell release (version 1.0.2)

If you're curious, the Wikipedia bug was caused by me forgetting to run the article title through `decodeURIcomponent()` first. Whoops. And the YouTube bug was caused by me only covering the case where the `?t=` timestamp was on `youtu.be` links, not `youtube.com`. Double whoops.

Big thanks again! :tada:
